code/modules/modular_computers/file_system/programs/security/implant_tracker.dm

```diff
@@ -0,0 +1,123 @@
+		ui.auto_update_content = TRUE
+		. = TRUE
+
+/datum/computer_file/program/implant_tracker/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	. = ..()
+	data = . || data || list()
+
+	// Gather data for computer header
+	var/headerdata = get_header_data(data["_PC"])
+	if(headerdata)
+		data["_PC"] = headerdata
+		. = data
+	
+	var/list/chem_implants = list()
+	for(var/obj/item/implant/chem/C in implants)
+		var/turf/Tr = get_turf(C)
+		if(!Tr || !isStationLevel(Tr.z))
+			continue
+		if(!C.implanted)
+			continue
+		var/list/chem_info = list(
+			"implanted_name" = C.imp_in.real_name,
+			"remaining_units" = round(C.reagents.total_volume, 0.1),
+			"ref" = "\ref[C]"
+			)
+		chem_implants[++chem_implants.len] = chem_info
+	VUEUI_SET_CHECK(data["chem_implants"], chem_implants, ., data)
+	var/list/tracking_implants = list()
+	for(var/obj/item/implant/tracking/T in implants)
+		var/turf/Tr = get_turf(T)
+		if(!Tr || !isStationLevel(Tr.z))
+			continue
+		if(!T.implanted)
+			continue
+		var/loc_display = "Unknown"
+		var/mob/living/carbon/M = T.imp_in
+		if(isStationLevel(M.z) && !istype(M.loc, /turf/space))
+			var/turf/mob_loc = get_turf(M)
+			loc_display = mob_loc.loc
+		if(T.malfunction)
+			loc_display = pick(teleportlocs)
+		var/list/tracker_info = list(
+			"id" = T.id,
+			"loc_display" = loc_display,
+			"ref" = "\ref[T]"
+			)
+		tracking_implants[++tracking_implants.len] = tracker_info
+	VUEUI_SET_CHECK(data["tracking_implants"], tracking_implants, ., data)
+
+	return data // This UI needs to constantly update
+
+
+/datum/computer_file/program/implant_tracker/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return TRUE
+
+	if(href_list["inject1"])
+		var/obj/item/implant/I = locate(href_list["inject1"]) in implants
+		if(I)
+			if(I.reagents.total_volume < 1)
+				to_chat(usr, SPAN_WARNING("\The [I] does not have enough of a payload to do this!"))
+				return
+			I.activate(1)
+
+	else if(href_list["inject5"])
+		var/obj/item/implant/I = locate(href_list["inject5"]) in implants
+		if(I)
+			if(I.reagents.total_volume < 5)
+				to_chat(usr, SPAN_WARNING("\The [I] does not have enough of a payload to do this!"))
+				return
+			I.activate(5)
+
+	else if(href_list["inject10"])
+		var/obj/item/implant/I = locate(href_list["inject10"]) in implants
+		if(I)
+			if(I.reagents.total_volume < 10)
+				to_chat(usr, SPAN_WARNING("\The [I] does not have enough of a payload to do this!"))
+				return
+			I.activate(10)
+
+	else if(href_list["warn"])
+		var/warning = sanitize(input(usr, "Message:", "Enter your message here!") as text|null)
+		if(!warning)
+			return
+
+		warning = capitalize(warning)
+		var/static/list/correct_punctuation = list("!" = TRUE, "." = TRUE, "?" = TRUE, "-" = TRUE, "~" = TRUE, "*" = TRUE, "/" = TRUE, ">" = TRUE, "\"" = TRUE, "'" = TRUE, "," = TRUE, ":" = TRUE, ";" = TRUE)
+		var/ending = copytext(warning, length(warning), (length(warning) + 1))
+		if(ending && !correct_punctuation[ending])
```

Comment:
This exists in code for like the 4th time now, would be best to make it a proc at some point.